src/NServiceBus.Core/Sagas/SagaMetadata.cs

```diff
@@ -378,13 +383,15 @@ static void ValidateMapping<TMessage>(Expression<Func<TMessage, object>> message
 
                 var propertyInfo = memberExpr.Member as PropertyInfo;
 
-                const string message = "Message properties mapped to the saga id needs to be of type Guid, please change property {0} on message {1} to a Guid";
+                const string message = "Saga mapping expressions are not the same types. {0}.{1} is of type {2}, while {3}.{4} is of type {5}.";
```

Comment:
Feels a bit technical, can we make it speak in terms of "saga" and "message"?

eg: `When mapping a message to a saga the types of the message member and the saga property must match. Xyz is of type and abc is of type` or something along those lines?